### PR TITLE
Java PEC: create PECs per arc ID; add callback to InitializeProxy method

### PR DESCRIPTION
**Pull request by mmandlis**

### CODE REVIEW

File: src/javaharness/java/arcs/api/PECInnerPortImpl.java [hunk 1]
```diff
@@ -1,27 +1,32 @@
...
+        this.pec = pec;
+        this.mapper = new ThingMapper("j");
+        this.jsonParser = jsonParser;
     }
 
     @Override
     public void handleMessage(PortableJson message) {
         String messageType = message.getString("messageType");
         PortableJson messageBody = message.getObject("messageBody");
-        String identifier = messageBody.getString("identifier");
```
Comment (cromwellian) [inline src/javaharness/java/arcs/api/PECInnerPortImpl.java]: Why did you move and inline this?

File: src/javaharness/java/arcs/api/PECInnerPortImpl.java [hunk 2]
```diff
@@ -37,29 +42,42 @@ public void handleMessage(PortableJson message) {
...
         }
     }
 
     @Override
-    public void InitializeProxy(StorageProxy storageProxy) {
-        // TODO: Implement.
-        shellApi.postMessage("InitializeProxy: " + storageProxy.id);
+    public void InitializeProxy(StorageProxy storageProxy, Function<PortableJson, Void> callback) {
+        PortableJson message = jsonParser.parse("{}");
+        message.put("messageType", "InitializeProxy");
```
Comment (cromwellian) [inline src/javaharness/java/arcs/api/PECInnerPortImpl.java]: just as an aside for future cleanup, in Java it is usually expected that string constants are pulled out into a public static final String CONSTANT = "constant" field and documented.

File: src/javaharness/java/arcs/api/StorageProxy.java [hunk 3]
```diff
@@ -28,10 +29,13 @@ public void register(NativeParticle particle, Handle handle) {
     observers.put(handle, particle);
 
     if (!listenerAttached) {
-      port.InitializeProxy(this); // TODO: pass `onSynchronize` callback to the port.
+      port.InitializeProxy(this, new Function<PortableJson, Void>() {
+          @Override public Void apply(PortableJson json) {
```
Comment (cromwellian) [inline src/javaharness/java/arcs/api/StorageProxy.java]: If you expect a method to take a parameter, but return nothing, don't use Function<Param, Result>, but use Consumer<ParamType> in the API (e.g. initializeProxy(Consumer<T>) }

File: src/javaharness/java/arcs/webimpl/DeviceClientJsImpl.java [hunk 4]
```diff
@@ -20,15 +22,16 @@
 
     private PortableJsonParser jsonParser;
     private Map<String, ArcsEnvironment.SuggestionListener> inProgress;
-    private PECInnerPort port;
+    private PECInnerPortFactory portFactory;
```
Comment (cromwellian) [inline src/javaharness/java/arcs/webimpl/DeviceClientJsImpl.java]: As an aside, in general, fields that are only ever assigned in constructors (e.g. const) should be marked final (e.g. private final PECInnerPortFactory portFactory)

File: src/javaharness/java/arcs/api/PECInnerPortImpl.java (hunk 1 above)
Comment (mmandlis) [reply src/javaharness/java/arcs/api/PECInnerPortImpl.java]: because SimpleCallback (and other future calls to come) don't have this field in the message.

File: src/javaharness/java/arcs/webimpl/DeviceClientJsImpl.java (hunk 4 above)
Comment (mmandlis) [reply src/javaharness/java/arcs/webimpl/DeviceClientJsImpl.java]: done (and actually found one redundant data member that i removed in PEC port impl)

File: src/javaharness/java/arcs/api/StorageProxy.java (hunk 3 above)
Comment (mmandlis) [reply src/javaharness/java/arcs/api/StorageProxy.java]: done, thanks!

File: src/javaharness/java/arcs/api/PECInnerPortImpl.java (hunk 2 above)
Comment (mmandlis) [reply src/javaharness/java/arcs/api/PECInnerPortImpl.java]: declared constants for all the strings in this file. however for now they are private, so i'll document them later :)
